retrofit/src/main/java/retrofit2/RequestBuilder.java

```diff
@@ -138,6 +134,17 @@ private static void canonicalize(Buffer out, String input, int pos, int limit,
     }
   }
 
+  private static boolean shouldSkipCodePoint(final int codePoint) {
```

Comment:
This is unrelated to XSS. The actual purpose is quite hilarious: it’s due to the fact that URLs are occasionally transcribed on paper or similar, and as a consequence spaces are not significant.